lifecycle/services/export.go

```diff
@@ -0,0 +1,174 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+	"net/url"
+)
+
+const (
+	exportReleaseBundleEndpoint       = "export"
+	exportReleaseBundleStatusEndpoint = "export/status"
+)
+
+type RlExportedStatus string
+
+const (
+	ExportCompleted    RlExportedStatus = "COMPLETED"
+	ExportProcessing   RlExportedStatus = "IN_PROGRESS"
+	ExportFailed       RlExportedStatus = "FAILED"
+	ExportNotTriggered RlExportedStatus = "NOT_TRIGGERED"
+)
+
+type ReleaseBundleExportedStatusResponse struct {
+	Status      RlExportedStatus `json:"status,omitempty"`
+	RelativeUrl string           `json:"relative_download_url,omitempty"`
+	DownloadUrl string           `json:"download_url,omitempty"`
+}
+
+type ReleaseBundleExportParams struct {
+	ReleaseBundleDetails
+	Modifications `json:"modifications"`
+}
+
+type RbExportBody struct {
+	ReleaseBundleExportParams
+}
+
+type exportStatusOperation struct {
+	reqBody     RbExportBody
+	queryParams CommonOptionalQueryParams
+}
+
+func (exs *exportStatusOperation) getOperationRestApi() string {
+	fullUrl, err := url.JoinPath(distributionBaseApi, exportReleaseBundleStatusEndpoint, exs.reqBody.ReleaseBundleName, exs.reqBody.ReleaseBundleVersion)
+	if err != nil {
+		panic(err)
+	}
+	return fullUrl
+}
+
+func (exs *exportStatusOperation) getRequestBody() any {
+	return exs.reqBody
+}
+
+func (exs *exportStatusOperation) getOperationSuccessfulMsg() string {
+	return "Successfully received Release Bundle export status"
+}
+
+func (exs *exportStatusOperation) getOperationParams() CommonOptionalQueryParams {
+	return exs.queryParams
+}
+
+func (exs *exportStatusOperation) getSigningKeyName() string {
+	return ""
+}
+
+type exportOperation struct {
+	reqBody     RbExportBody
+	queryParams CommonOptionalQueryParams
+}
+
+func (exp *exportOperation) getOperationRestApi() string {
+	fullUrl, err := url.JoinPath(distributionBaseApi, exportReleaseBundleEndpoint, exp.reqBody.ReleaseBundleName, exp.reqBody.ReleaseBundleVersion)
+	if err != nil {
+		panic(err)
+	}
+	return fullUrl
+}
+
+func (exp *exportOperation) getRequestBody() any {
+	return exp.reqBody
+}
+
+func (exp *exportOperation) getOperationSuccessfulMsg() string {
+	return "Release Bundle successfully exported"
+}
+
+func (exp *exportOperation) getOperationParams() CommonOptionalQueryParams {
+	return exp.queryParams
+}
+
+func (exp *exportOperation) getSigningKeyName() string {
+	return ""
+}
+
+func (rbs *ReleaseBundlesService) ExportReleaseBundle(rlExportParams *ReleaseBundleExportParams, queryParams CommonOptionalQueryParams) (exportResponse ReleaseBundleExportedStatusResponse, err error) {
+	// Check the current status
+	if exportResponse, err = rbs.getExportedReleaseBundleStatus(rlExportParams, queryParams); err != nil {
+		return
+	}
+	// Trigger export if needed
```

Comment:
We can simplify the code a bit by checking here.
```
if exportResponse.Status == ExportCompleted {
    return
}
```
Then the last if is not necessary.